src/helpers/markdown.js

```diff
@@ -91,5 +91,18 @@ export const getTableOfContent = (source) => {
   const tokens = marked.lexer(source)
   // remove the first 2 tokens to avoid meta information
   // and then return every tokens that is of header type
-  return tokens.slice(2).filter(token => token.type === "heading")
+  const headers = tokens.slice(2).filter(token => token.type === "heading" && token.depth <= 4)
+  headers.forEach(header => {
+    const html = marked.parser([header])
+    const doc = new DOMParser().parseFromString(html, "text/xml")
+    header.id = doc.firstChild.id
+    header.actualText = html
+      .replace(/<[^>]*>/g, "")
+      .replace(/&lt;/g, "<")
+      .replace(/&gt;/g, ">")
+      .replace(/&quot;/g, "\"")
+      .replace(/&#39;/g, "'")
+      .replace(/&amp;/g, "&")
```

Comment:
We can also utilize the `doc`. Check if this works well.
```suggestion
    header.text = doc.firstChild.textContent
```